libs/blocks/locui/actions/index.js

```diff
@@ -223,16 +223,24 @@ export function closeActionModal() {
   document.querySelector('.dialog-modal').dispatchEvent(new Event('closeModal'));
 }
 
+let isSyncing = false;
 export async function syncFragsLangstore() {
+  if (isSyncing) return;
+  isSyncing = true;
   closeActionModal();
   if (syncFragments.value?.length) {
     await syncToExcel(syncFragments.value);
     syncFragments.value = [];
   }
   await syncToLangstore();
+  isSyncing = false;
```

Comment:
Maybe I'm missing something, but `isSyncing = true;` seems redundant in here as we're setting `isSyncing` to `false` at the end of the function 🤔

Reply:
Yes, I am doing an early return if the function is currently awaiting a response from the backend and then after the async function has finished all its processes the function is then re-enabled. This is meant to stop the UI from sending an additional request to the backend when a user double clicks the button.